File: go-ms/domain/model/hello/gateways/hello_repository.go
package gateways

type HelloRepository interface {
	HelloHttp(latency string) (string, error)
	HelloHttps(latency string) (string, error)
	HelloConnectionPoolHttp1(latency string) (string, error)
	HelloConnectionPoolHttp2(latency string) (string, error)
}

File: java-reactive-ms/domain/model/src/main/java/com/maocq/model/hello/gateways/HelloRepository.java
package com.maocq.model.hello.gateways;

import reactor.core.publisher.Mono;

public interface HelloRepository {

    Mono<String> helloHttp(int latency);
    Mono<String> helloHttps(int latency);
    Mono<String> helloConnectionPoolHttp1(int latency);
    Mono<String> helloConnectionPoolHttp2(int latency);
}
